list-navigation/lambda/handlers/record-inital-event.ts

```diff
@@ -29,6 +30,11 @@ export class RecordInitialEventHandler extends BaseApiHandler {
             // reset page tokens to ensure we start navigating list at beginning, as it's 
             // possible a custom user event was just triggered to restart navigation
             // of a list that was setup before and already navigated some
+
+            //Stack gets initialized, in case of DDBListProvider
+            if(ListNav.getProvider(sessionState.activeList).getName() == DDBListProvider.NAME){
```

Comment:
we should not be hard-coding extra logic into the handlers for a specific type of list provider; the whole point of the list provider interface is to provide a layer of abstraction, the handlers should have no idea how the provider works and the provider should have no idea how the handlers work; both only know the interface between

Reply:
Got it! Will try to shift the whole thing to the DDBListProvider. Thanks!